lib/SILOptimizer/Transforms/PhiArgumentOptimizations.cpp

```diff
@@ -111,7 +122,18 @@ bool RedundantPhiEliminationPass::optimizeArgs(SILBasicBlock *block) {
         continue;
 
       if (valuesAreEqual(arg1, arg2)) {
-        arg2->replaceAllUsesWith(arg1);
+        SILValue replaceWith = arg1;
+        // @owned phi args can only be equal if all the incoming values had a
+        // None ownership. To replace, create a copy_value of the duplicate arg.
+        if (arg1->getOwnershipKind() == OwnershipKind::Owned) {
+          assert(arg2->getOwnershipKind() == OwnershipKind::Owned);
```

Comment:
How do we know for sure that `arg2` doesn't have `OwnershipKind::None` for example?

Reply:
Thanks, I had missed that case. Added support for it now.